src/System.Drawing.Common/src/System/Drawing/Printing/PrinterSettings.cs

```diff
@@ -215,41 +215,26 @@ public static StringCollection InstalledPrinters
             get
             {
                 int returnCode;
-                int bufferSize;
-                int count;
-                int level, sizeofstruct;
-                // Note: Level 5 doesn't seem to work properly on NT platforms
-                // (atleast the call to get the size of the buffer reqd.),
-                // and Level 4 doesn't work on Win9x.
-                //
-                if (Environment.OSVersion.Platform == System.PlatformID.Win32NT)
+                int sizeofstruct;
+                // Note: The call to get the size of the buffer required for level 5 does not work property on NT platforms.
+                const int Level = 4;
+                // PRINTER_INFO_4 are 12 bytes in size
+                if (IntPtr.Size == 8)
                 {
-                    level = 4;
-                    // PRINTER_INFO_4 are 12 bytes in size
-                    if (IntPtr.Size == 8)
-                    {
-                        sizeofstruct = (IntPtr.Size * 2) + (Marshal.SizeOf(typeof(int)) * 1) + PADDING_IA64;
-                    }
-                    else
-                    {
-                        sizeofstruct = (IntPtr.Size * 2) + (Marshal.SizeOf(typeof(int)) * 1);
-                    }
+                    sizeofstruct = (IntPtr.Size * 2) + (Marshal.SizeOf(typeof(int)) * 1) + PADDING_IA64;
                 }
                 else
                 {
-                    level = 5;
-                    // PRINTER_INFO_5 are 20 bytes in size
-                    sizeofstruct = (IntPtr.Size * 2) + (Marshal.SizeOf(typeof(int)) * 3);
+                    sizeofstruct = (IntPtr.Size * 2) + (Marshal.SizeOf(typeof(int)) * 1);
```

Comment:
Yeah, I agree.

What will be nice is to move some of the PInvoke code to the Interop.Gdip.... format that we have across the code base. This would allow us to share code with Unix, and cleanup the interop code while we're at it on a case-by-case basic